ui/page/transaction/transactions_page.go

```diff
@@ -209,99 +276,159 @@ func (pg *TransactionsPage) loadTransactions(offset, pageSize int32) ([]*sharedW
-									if pg.scroll.ItemsCount() == 0 {
-										padding := values.MarginPadding16
-										txt := pg.Theme.Body1(values.String(values.StrNoTransactions))
-										txt.Color = pg.Theme.Color.GrayText3
-										gtx.Constraints.Min.X = gtx.Constraints.Max.X
-										return layout.Center.Layout(gtx, func(gtx C) D {
-											return layout.Inset{Top: padding, Bottom: padding}.Layout(gtx, txt.Layout)
-										})
-									}
+	layoutContents := []layout.StackChild{layout.Expanded(func(gtx C) D {
+		return layout.Inset{Top: values.MarginPadding60}.Layout(gtx, func(gtx C) D {
+			itemCount := pg.scroll.ItemsCount()
+			card := pg.Theme.Card()
+			// return "No transactions yet" text if there are no transactions
+			if itemCount == 0 {
+				padding := values.MarginPadding16
+				txt := pg.Theme.Body1(values.String(values.StrNoTransactions))
+				txt.Color = pg.Theme.Color.GrayText3
+				return card.Layout(gtx, func(gtx C) D {
+					gtx.Constraints.Min.X = gtx.Constraints.Max.X
+					return layout.Center.Layout(gtx, func(gtx C) D {
+						return layout.Inset{Top: padding, Bottom: padding}.Layout(gtx, txt.Layout)
+					})
+				})
+			}
 
-									wallTxs := pg.scroll.FetchedData()
-									return pg.transactionList.Layout(gtx, len(wallTxs), func(gtx C, index int) D {
-										tx := wallTxs[index]
-										return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
-											layout.Rigid(func(gtx C) D {
-												return components.LayoutTransactionRow(gtx, pg.Load, wal, tx, true)
-											}),
-											layout.Rigid(func(gtx C) D {
-												// No divider for last row
-												if index == len(wallTxs)-1 {
-													return layout.Dimensions{}
-												}
+			return pg.scroll.List().Layout(gtx, 1, func(gtx C, i int) D {
+				return layout.Inset{Right: values.MarginPadding2}.Layout(gtx, func(gtx C) D {
+					return card.Layout(gtx, func(gtx C) D {
+						if itemCount == -1 || pg.showLoader {
+							gtx.Constraints.Min.X = gtx.Constraints.Max.X
+							return layout.Center.Layout(gtx, pg.materialLoader.Layout)
+						}
 
-												gtx.Constraints.Min.X = gtx.Constraints.Max.X
-												separator := pg.Theme.Separator()
-												return layout.E.Layout(gtx, func(gtx C) D {
-													// Show bottom divider for all rows except last
-													return layout.Inset{Left: values.MarginPadding56}.Layout(gtx, separator.Layout)
-												})
-											}),
-										)
+						wallTxs := pg.scroll.FetchedData()
+						return pg.transactionList.Layout(gtx, len(wallTxs), func(gtx C, index int) D {
+							tx, wal := pg.txAndWallet(wallTxs[index])
+							return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+								layout.Rigid(func(gtx C) D {
+									isHiddenAssetsInfo := true
+									if pg.selectedWallet == nil {
+										isHiddenAssetsInfo = !isHiddenAssetsInfo
+									}
+									return components.LayoutTransactionRow(gtx, pg.Load, wal, tx, isHiddenAssetsInfo)
```

Comment:
What's this boolean that's being passed to `components.LayoutTransactionRow`? The fn says it accepts a `isTxPage bool` (which I don't even know what it means) but here you're passing `isHiddenAssetsInfo` and somewhere else you pass `showAssets`. What does this boolean variable really control?

Reply:
The bool is an indicator for the tx list to show some additional information about the tx.. example the wallet it belongs to.